src/backends/cpu.jl

```diff
@@ -41,7 +41,15 @@ end
 function __run(obj, ndrange, iterspace, args, dependencies)
     return Threads.@spawn begin
         if dependencies !== nothing
-            Base.sync_end(map(e->e.task, dependencies))
+            cpu_tasks = Core.Task[]
+            for event in dependencies
```

Comment:
In general I would invert this scheme to first wait on a CPU events, and then wait on the GPU events.
The CPU wait is more efficiently implemented in the scheduler and we don't have to busy-wait on the GPU.

Reply:
Okay, I did this and rearranged the cuda backed to first enqueue the gpu waits and then wait on the CPU.